galleries/examples/scales/symlog_demo.py

```diff
@@ -34,12 +39,85 @@
 plt.show()
 
 # %%
-# It should be noted that the coordinate transform used by ``symlog``
-# has a discontinuous gradient at the transition between its linear
-# and logarithmic regions. The ``asinh`` axis scale is an alternative
-# technique that may avoid visual artifacts caused by these discontinuities.
+# Linear threshold
+# ----------------
+# Since each decade on a logarithmic scale covers the same amount of visual space
+# and there are infinitely many decades between a given number and zero, the symlog
+# scale must deviate from logarithmic mapping in a small range (-x0, x0), so that
+# that range is mapped to a finite visual space.
+#
+# The symlog scale achieves this by defining a parameter *linthresh* and switching
+# to a linear mapping in the region *(-linthresh, linthresh)*.
+
+
+def format_axes(ax, title=None):
+    """A helper function to better visualize properties of the symlog scale."""
+    ax.xaxis.get_minor_locator().set_params(subs=[2, 3, 4, 5, 6, 7, 8, 9])
+    ax.grid()
+    ax.xaxis.grid(which='minor')  # minor grid on too
+    linthresh = ax.xaxis.get_transform().linthresh
+    linscale = ax.xaxis.get_transform().linscale
+    ax.axvspan(-linthresh, linthresh, color='0.9')
+    if title:
+        ax.set_title(title.format(linthresh=linthresh, linscale=linscale))
+
+
+x = np.linspace(-60, 60, 201)
+y = np.linspace(0, 100.0, 201)
+
+fig, (ax1, ax2) = plt.subplots(nrows=2, layout="constrained")
+
+ax1.plot(x, y)
+ax1.set_xscale('symlog', linthresh=1)
+format_axes(ax1, title='Linear region: linthresh={linthresh}')
+
+ax2.plot(x, y)
+ax2.set_xscale('symlog', linthresh=5)
+format_axes(ax2, title='Linear region: linthresh={linthresh}')
 
 # %%
+# Generally, *linthresh* should be chosen so that no or only a few
+# data points are in the linear region. As a rule of thumb,
+# :math:`linthresh \approx \mathrm{min} |x|`.
```

Comment:
Where does this rule of thumb come from? I feel like this might be very data dependent and perhaps we shouldn't suggest anything 🤷

Reply:
This comes from the two arguments
- linthresh should not be much larger than your min value, as that would push too many points into the linear region
- linthresh should not be much smaller than your min value, as that would mean you have empty decades between linthresh and your data - effectively pushing your data to the borders of the plot and leaving a lot of unused space in the center.

Unfortunately, one has to adapt linthresh depending on the data to get a decent looking plot. When I looked into this first I had no idea what value to choose. While you can create datasets that needs special different tuning, I feel the approximate rule of thumb will generally be a good starting point..